kdc/fast.c

```diff
@@ -560,6 +568,83 @@ fast_unwrap_request(astgs_request_t r,
 	free_AP_REQ(&ap_req);
 	if (ret)
 	    goto out;
+
+	ret = krb5_unparse_name(r->context, armor_server,
+				&armor_server_principal_name);
+	if (ret)
+	    goto out;
+
+	ret = _kdc_verify_flags(r->context, r->config,
+				&ticket->ticket,
+				armor_server_principal_name);
+	free(armor_server_principal_name);
+	if (ret) {
+	    _kdc_audit_addreason((kdc_request_t)r,
+				 "Armor TGT expired or invalid");
+	    goto out;
+	}
+
+	if (tgs_ticket != NULL)
+	    flags = HDB_F_FOR_TGS_REQ;
+	else
+	    flags = HDB_F_FOR_AS_REQ;
+	if (r->req.req_body.kdc_options.canonicalize)
+	    flags |= HDB_F_CANON;
+
+	/* Fetch the name from the TGT. */
+	ret = _krb5_principalname2krb5_principal(r->context,
+						 &fast_principal,
+						 ticket->ticket.cname,
+						 ticket->ticket.crealm);
+	if (ret)
+	    goto out;
+
+	/* Look up the name given in the TGT in the database. */
```

Comment:
Some of these comments that are explained by the code itself can probably be removed.